### DISCUSSION TITLE
Collecting analytics about macOS versions for install, upgrade, outdated & similar requests

### DISCUSSION THREAD

**Discussion by rgoldberg** [Homebrew/discussions#5651] [Everyday usage] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I'm working on a macOS-only Swift project is available as a Homebrew formula.

It is marked as compatible back to macOS 10.11 in its build settings.

It would be nice to eventually use some Swift features only available in macOS 10.15 & newer.

I want to determine how many users use macOS < 10.15.

Homebrew installation/upgrade/outdated requests analytics could be useful, if the necessary info is collected.

I’ve seen the `brew info —analytics` for the formula, but that doesn’t include the macOS versions for the installs/errors. Is macOS version info collected?

Is any info collected about `brew outdated` requests, `brew update` requests, or actual `brew update` updates? I’d like to know the numbers for every macOS version.

If none of this is collected, is there any way that Homebrew could be updated such that it could be collected?

**Comment by SMillerDev** [comment]
Since brew doesn't support anything pre macOS 12, I don't know how well those analytics would work for your purpose.

All analytics data is available at: https://formulae.brew.sh/analytics/

**Reply by rgoldberg** [reply]
Thanks for the info.

The per-OS-version analytics for the last year include events all the way back to macOS 10.11, so, while brew doesn't officially support macOS 12 older, some people still use it on such systems.

Are any of the events recorded and/or made available to the public on a per-OS-version-&-formula basis, or are they just available recorded and/or made available to the public separately in per-OS-version by itself & per-formula by itself?

If data is available in a more granular manner than is show on the web site, is it possible to access it?

Are upgrade / outdated request analytics recorded? I don't see them anywhere. Could they be recorded as installs, even though they're really upgrades?

**Reply by MikeMcQuaid** [reply, marked as answer]
> Are any of the events recorded and/or made available to the public on a per-OS-version-&-formula basis, or are they just available recorded and/or made available to the public separately in per-OS-version by itself & per-formula by itself?
> 
> If data is available in a more granular manner than is show on the web site, is it possible to access it?

This event data could perhaps be extracted on a per-formula basis but this isn't something we've had requested before and would be a non-trivial lift for us on the InfluxDB side.

Nothing else is available publicly/more granularly other than what's on formulae.brew.sh

> Are upgrade / outdated request analytics recorded? I don't see them anywhere. Could they be recorded as installs, even though they're really upgrades?

They are counted as installs.
